packages/medusa/src/utils/repository.ts

```diff
@@ -163,3 +165,66 @@ export function mergeEntitiesWithRelations<T>(
     merge({}, ...entityAndRelations)
   )
 }
+
+/**
+ * Apply the appropriate order depending on the requirements
+ * @param repository
+ * @param order
+ * @param qb
+ * @param alias
+ * @param shouldJoin
+ */
+export function applyOrdering<T>({
+  repository,
+  order,
+  qb,
+  alias,
+  shouldJoin,
+}: {
+  repository: Repository<T>
+  order: Record<string, "ASC" | "DESC">
+  qb: SelectQueryBuilder<T>
+  alias: string
+  shouldJoin: (relation: string) => boolean // In case a join is applied elsewhere and therefore should not be re applied here
+}) {
+  const toSelect: string[] = []
+  const parsed = Object.entries(order).reduce(
+    (acc, [orderPath, orderDirection]) => {
+      if (orderPath.includes(".")) {
+        const relationsToJoin = orderPath.split(".")
+        const propToOrder = relationsToJoin.pop()
+
+        relationsToJoin.reduce(
+          ([parent, parentMetadata], child) => {
+            const relationMetadata = (
+              parentMetadata as EntityMetadata
+            ).relations.find(
+              (relationMetadata) => relationMetadata.propertyName === child
+            )
+            const shouldApplyJoin = shouldJoin(child)
+            if (shouldApplyJoin) {
+              qb.leftJoin(`${parent}.${relationMetadata!.propertyPath}`, child)
+            }
+            return [child, relationMetadata!.inverseEntityMetadata]
+          },
+          [alias, repository.metadata]
+        )
```

Comment:
suggestion(non-blocking): Though this is super nice, it is also pretty hard to read. I think we should add a couple of comments in case we need to revisit it.

Reply:
We could include an example of parsing `order` in the JSDoc?